src/sentry/insights/migrations/0001_add_starred_transactions_model.py

```diff
@@ -0,0 +1,70 @@
+# Generated by Django 5.1.7 on 2025-04-10 13:12
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+import sentry.db.models.fields.bounded
+import sentry.db.models.fields.foreignkey
+import sentry.db.models.fields.hybrid_cloud_foreign_key
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production.
+    # This should only be used for operations where it's safe to run the migration after your
+    # code has deployed. So this should not be used for most operations that alter the schema
+    # of a table.
+    # Here are some things that make sense to mark as post deployment:
+    # - Large data migrations. Typically we want these to be run manually so that they can be
+    #   monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   run this outside deployments so that we don't block them. Note that while adding an index
+    #   is a schema change, it's completely safe to run the operation after the code has deployed.
+    # Once deployed, run these manually via: https://develop.sentry.dev/database-migrations/#migration-deployment
+
+    is_post_deployment = False
+
+    initial = True
+
+    dependencies = [
+        ("sentry", "0865_file_offsets"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="InsightsStarredSegment",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                ("date_updated", models.DateTimeField(auto_now=True)),
+                ("date_added", models.DateTimeField(auto_now_add=True)),
+                (
+                    "user_id",
+                    sentry.db.models.fields.hybrid_cloud_foreign_key.HybridCloudForeignKey(
+                        "sentry.User", db_index=True, on_delete="CASCADE"
+                    ),
+                ),
+                ("segment_name", models.CharField(max_length=255)),
```

Comment:
Couldn't segments have longer names, or are they capped at 255?

Reply:
Great question, the previous version of this feature (the team based favourited transactions), had a max limit of 200. But, i think we should still double check this limit for EAP spans.

Reply:
@markstory so it looks like there is no limit on the storage of segment_name, any objections to just matching that?